rpcclient/balancer.go

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) 2015-2018 The Decred developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+package rpcclient
+
+import (
+	"github.com/gorilla/websocket"
+)
+
+// Balancer defines requirements for a load balancer
+// using list of hostaddresses as the source.
+type Balancer interface {
+	// NextConn gets the next usable connection.
+	// If the methodName is one of notifications, then would remember the connection used
+	// for this method and would next time pick the same connection.
+	// Else would consider normal roundrobin routine to get the next connection.
+	NextConn(methodName string) (wsConn *websocket.Conn, hostAddress *HostAddress, err error)
+
+	// ConnectionState gets the connectionstate of corresponding wsConn.
+	ConnectionState(host string) (state ConnectionState, ok bool)
+
+	// WsConnection gets the wsConn corresponding to the host.
+	WsConnection(host string) (wsConn *websocket.Conn, connOk bool)
+
+	// AllDisconnectedWsConns will iterate over the connection state map and
+	// return all the wsConnections that have their state as Disconnected.
+	AllDisconnectedWsConns() []*HostAddress
+
+	// NotifyConnStateChange is called by rpcClient when the ConnectionState
+	// for a hostaddress changes.
+	NotifyConnStateChange(sc *HostAddress, state ConnectionState, sync bool)
+
+	// UpdateReconnectAttempt will increase retryattempt + 1.
```

Comment:
You're referring to a variable name that does not exist here: `retryattempt`, you probably meant `retryCount`. I'd suggest not using variable names in comments, it's easy to lose track of outdated comments when var names are modified. Plus a comment like `// UpdateReconnectAttempt increments the connection's retry counter by one.`  reads better imo.